types/file-exists/index.d.ts

```diff
@@ -3,13 +3,25 @@
 // Definitions by: BendingBender <https://github.com/BendingBender>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+/// <reference types="node" />
+
 export = fileExists;
 
-declare function fileExists(filepath: string, callback?: (err: Error | null, exists: boolean) => void): void;
-declare function fileExists(filepath: string, options?: fileExists.Options, callback?: (err: Error | null, exists: boolean) => void): void;
+type FilePathType = string | Buffer | URL;
```

Comment:
This can't be correct. `filepath` is passed directly to `path.join` (https://github.com/scottcorgan/file-exists/blob/master/index.js#L54) when the `root` option is set. You'll need an overload when this option is set that only accepts a `string` as `filePath`.

Reply:
You're absolutely right, good catch. Looking at that line, it looks as though `filepath` must be a string if `root` on the options object is a non-empty string. I've committed some types that should capture that behavior.